enhancements/insights/config-api.md

```diff
@@ -0,0 +1,245 @@
+---
+title: defining-new-insights-config-api
+authors:
+  - "@tremes"
+reviewers:
+  - "@bparees"
+  - "@deads2k"
+  - "@mfojtik"
+approvers:
+  - "@bparees"
+  - "@deads2k"
+  - "@mfojtik"
+creation-date: 2022-02-16
+last-updated: 2022-03-24
+status: implementable
+see-also:
+  - "https://issues.redhat.com/browse/CCXDEV-6852"
+  - "https://issues.redhat.com/browse/CCX-195"
+---
+
+# Define and expose Insights configuration API
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Graduation criteria for dev preview, tech preview, GA
+- [x] User-facing documentation is created/updated in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement extends the `config.openshift.io` API with a new Insights API. The new API will introduce new configuration options (including an option to force the data gathering as well as to disable it completely) as well as status reporting of the Insights operator. 
+
+## Motivation
+
+The main motivation for this new API is to allow users to refresh (or force) the gathering of the Insights data (see the [User stories](#user-stories) section for more details) as well as to disable the gathering completely. This new API will allows us to report various Insights status reports (including the Insights analysis report that is now exposed via Prometheus metrics).
+
+### Goals
+
+The goal of this enhancement is to propose a basic structure for a new Insights configuration API and define new configuration options. This includes following:
+
+- Define a new configuration option providing a way to trigger on demand Insights gathering
+- Define a new configuration option providing a way to include/exclude particular Insights data gatherers
+- Provide Insights status & reporting in the status field in the new API
+
+### Non-Goals
+
+- Move the majority of the current configuration options to a new configuration API - Insights has quite a lot of configuration options
+  and it does not make sense to expose everything in the new configuration API 
+- Removing the original configuration via `support` secret
+
+## Proposal
+
+The proposal is to introduce a new Insights `CustomResourceDefinition` in the OpenShift API providing a way to configure the Insights operator and to report its status. The new CRD should follow a common CRD structure providing `spec` and `status` attributes. 
+
+```go
+type InsightsSpec struct {
+	// Gathering spec attribute includes all the configuration options related to
+	// gathering of the Insights archive and its uploading to the ingress.
+	GatheringConfig GatheringConfig `json:"gatheringConfig,omitempty"`
+}
+
+type InsightsStatus struct {
+	// GatheringStatus provides basic information about the last Insights gathering.
+	GatheringStatus GatheringStatus `json:"gatheringStatus,omitempty"`
```

Comment:
`omitempty` doesn't do anything for non-pointer struct properties (golang/go#11939).  If you want this to be `omitempty`, you need `*GatheringStatus`.  And you probably do want a pointer anyway, because "there haven't been any gathers yet" seems like a valid thing.

Reply:
Updated. Thank you